Underscore.cfc

```diff
@@ -1940,12 +1940,12 @@ component {
 	/*
 		Returns a wrapped object. Calling methods on this object will continue to return wrapped objects until value is used.
 	*/
-	// TODO: make this work
-	public any function chain(obj) {
-		var _obj = new Underscore(arguments.obj);
-		return _.wrap(_obj, function (func) {
-			return new Underscore(func(arguments));
-		});
+	public any function chain(any obj) {
+		if (! structKeyExists(arguments, "obj")) {
+			arguments.obj = this.obj;
+		}
+
+		return new UnderscoreWrapper(arguments.obj, this);
```

Comment:
The library is designed to be imported as a standalone .cfc file, so I'd prefer not to introduce additional files that it depends upon. Could we create a new component on the fly and decorate it with the necessary methods instead? I'm not 100% sure if `onMissingMethod` would work with that pattern, but it might.